ecs-agent/api/resource/resource_validation.go

```diff
@@ -17,9 +17,9 @@ import (
 	"github.com/pkg/errors"
 )
 
-// ValidateResource checks if the provided resource type is valid, as well as if the attachment
+// ValidateResourceByResourceType checks if the provided resource type is valid, as well as if the attachment
 // properties of the specified resource are valid.
-func ValidateResource(resourceAttachmentProperties map[string]string) error {
+func ValidateResourceByResourceType(resourceAttachmentProperties map[string]string) error {
```

Comment:
Q: will we also add `AmazonElasticBlockStorage` as one of the switch cases in this function in a follow up PR?

Reply:
Nope, we will save AttachmentType in the `ResourceAttachment` and then use that to branch off different attachments.

Reply:
Oh I see, thanks for the clarification!